src/jit/codegencommon.cpp

```diff
@@ -10461,38 +10461,24 @@ unsigned CodeGen::getFirstArgWithStackSlot()
 {
 #if defined(UNIX_AMD64_ABI) || defined(_TARGET_ARMARCH_)
     unsigned baseVarNum = 0;
-#if defined(FEATURE_UNIX_AMR64_STRUCT_PASSING)
-    baseVarNum = compiler->lvaFirstStackIncomingArgNum;
-
-    if (compiler->lvaFirstStackIncomingArgNum != BAD_VAR_NUM)
+    // Iterate over all the local variables in the Lcl var table.
+    // They contain all the implicit arguments - thisPtr, retBuf,
+    // generic context, PInvoke cookie, var arg cookie,no-standard args, etc.
+    LclVarDsc* varDsc = nullptr;
+    for (unsigned i = 0; i < compiler->info.compArgsCount; i++)
     {
-        baseVarNum = compiler->lvaFirstStackIncomingArgNum;
-    }
-    else
-#endif // FEATURE_UNIX_ARM64_STRUCT_PASSING
-    {
-        // Iterate over all the local variables in the Lcl var table.
-        // They contain all the implicit arguments - thisPtr, retBuf,
-        // generic context, PInvoke cookie, var arg cookie,no-standard args, etc.
-        LclVarDsc* varDsc = nullptr;
-        for (unsigned i = 0; i < compiler->info.compArgsCount; i++)
-        {
-            varDsc = &(compiler->lvaTable[i]);
+        varDsc = &(compiler->lvaTable[i]);
 
-            // We are iterating over the arguments only.
-            assert(varDsc->lvIsParam);
+        // We are iterating over the arguments only.
+        assert(varDsc->lvIsParam);
```

Comment:
With /GSChecks we can make a "shadow" copy of an incoming struct argument.
Basically we are copying it from the incoming Arg Space into a new struct local that  is placed so that
it will catch any buffer overruns into its storage.   Something to think about with this change.

Reply:
Thanks, Brian. Just to make sure I understand - this would be a "new" local though, right? So that the original varDsc would still have the original offset. If that's the case, then I don't think that would impact this. Otherwise, this would have a problem because it assumes that they are laid out in order.

Reply:
Iy is a new local, I'm not sure about all the detaisl , but the IL reference to the arg are rewrtten to refer to the new local.  (There was a bug where we struct promoted it and end up with reference to both the original copy and to the individual promote fields.)

A fix and test case for the issue is here:
https://github.com/dotnet/coreclr/pull/17329